lib_test/test_fs_unix.ml

```diff
@@ -106,9 +108,22 @@ let read_zero_bytes () =
       sure %s is present in the test filesystem" content_file)
   | `Error e -> assert_fail e
 
+(* reasonable responses are `Ok [] or an error *)
+let read_negative_bytes () =
+  connect_or_fail () >>= fun fs ->
+  FS_impl.read fs content_file 0 (-5) >>= function
+  | `Ok [] -> Lwt.return_unit
+  | `Ok bufs ->
+    OUnit.assert_failure "reading -5 bytes from a non-empty file returned some cstructs"
+  | `Error (`No_directory_entry _) ->
+    OUnit.assert_failure (Printf.sprintf "read failed for a present file; please make
+      sure %s is present in the test filesystem" content_file)
+  | `Error (`Unknown_error _) -> Lwt.return_unit
```

Comment:
`Do different systems do this` - I have to confess to not remembering.  On looking at this code again (thanks for looking and commenting, by the way -- I'd completely forgotten about this PR), I'd much rather have some type constraint that forces the argument to `read` to be non-negative.  Failing that, a consistent and pattern-matchable error behavior would be nice; this particular case makes me think about a hierarchy of errors (giving `read` a negative length is probably more "severe" than the directory not existing, and if we can only report one error we should probably report that one).  `Unknown_error` with a string is rather unsatisfying.